core/src/main/java/com/linecorp/armeria/server/file/NonExistentHttpFile.java

```diff
@@ -59,17 +63,22 @@ public CompletableFuture<HttpResponse> read(Executor fileReadExecutor, ByteBufAl
     @Override
     public HttpService asService() {
         return (ctx, req) -> {
-            switch (req.method()) {
-                case HEAD:
-                case GET:
-                    if (location == null) {
-                        return HttpResponse.of(HttpStatus.NOT_FOUND);
-                    } else {
-                        return HttpResponse.ofRedirect(location);
-                    }
-                default:
-                    return HttpResponse.of(HttpStatus.METHOD_NOT_ALLOWED);
+            final HttpMethod method = req.method();
+            if (method != HttpMethod.GET && method != HttpMethod.HEAD) {
+                return HttpResponse.of(HttpStatus.METHOD_NOT_ALLOWED);
             }
+
+            if (location == null) {
+                return HttpResponse.of(HttpStatus.NOT_FOUND, MediaType.PLAIN_TEXT_UTF_8,
+                                       "Location is not set.");
+            }
```

Comment:
It is weird to send `"Location is not set."` for `HttpFile.nonExistent()`.
Should we check `isRedirect` first and just return `NOT_FOUND` without the message?
```java
if (isRedirect) {
    return HttpResponse.ofRedirect(location);
}

if (location == null) {
    return HttpResponse.of(HttpStatus.NOT_FOUND);
} else {
   ...
}
```